mitosheet/mitosheet/mito_widget.py

```diff
@@ -97,20 +99,46 @@ def update_shared_state_variables(self) -> None:
         self.sheet_data_json = self.steps_manager.sheet_data_json
         self.analysis_data_json = self.steps_manager.analysis_data_json
         self.user_profile_json = json.dumps({
-            # Dynamic, update each time
-            'userEmail': get_user_field(UJ_USER_EMAIL),
-            'receivedTours': get_user_field(UJ_RECEIVED_TOURS),
-            'receivedChecklists': get_user_field(UJ_RECEIVED_CHECKLISTS),
-            'isPro': is_pro(),
-            'telemetryEnabled': telemetry_turned_on(),
-            # Static over a single analysis
-            'pythonVersion': get_python_version(),
-            'pandasVersion': get_pandas_version(),
-            'isLocalDeployment': self.is_local_deployment,
-            'shouldUpgradeMitosheet': self.should_upgrade_mitosheet,
-            'numUsages': self.num_usages,
-            'mitoConfig': self.mito_config.get_mito_config()
-        })
+                # Dynamic, update each time
+                'userEmail': get_user_field(UJ_USER_EMAIL),
+                'receivedTours': get_user_field(UJ_RECEIVED_TOURS),
+                'receivedChecklists': get_user_field(UJ_RECEIVED_CHECKLISTS),
+                'isPro': is_pro(),
+                'telemetryEnabled': telemetry_turned_on(),
+                # Static over a single analysis
+                'pythonVersion': get_python_version(),
+                'pandasVersion': get_pandas_version(),
+                'isLocalDeployment': self.is_local_deployment,
+                'shouldUpgradeMitosheet': self.should_upgrade_mitosheet,
+                'numUsages': self.num_usages,
+                'mitoConfig': self.mito_config.get_mito_config()
+            })
+
+
+    def get_shared_state_variables(self) -> Dict[str, Any]:
+        """
+        Helper function for updating all the variables that are shared
+        between the backend and the frontend through trailets.
+        """
+        return {
+            'sheet_data_json': self.steps_manager.sheet_data_json,
+            'analysis_data_json': self.steps_manager.analysis_data_json,
+            'user_profile_json': json.dumps({
+                # Dynamic, update each time
+                'userEmail': get_user_field(UJ_USER_EMAIL),
+                'receivedTours': get_user_field(UJ_RECEIVED_TOURS),
+                'receivedChecklists': get_user_field(UJ_RECEIVED_CHECKLISTS),
+                'isPro': is_pro(),
+                'telemetryEnabled': telemetry_turned_on(),
+                # Static over a single analysis
+                'pythonVersion': get_python_version(),
+                'pandasVersion': get_pandas_version(),
+                'isLocalDeployment': self.is_local_deployment,
+                'shouldUpgradeMitosheet': self.should_upgrade_mitosheet,
+                'numUsages': self.num_usages,
+                'mitoConfig': self.mito_config.get_mito_config()
```

Comment:
Since this perfectly matches the user json code above, can we create a function for get_user_json() that we use in both places?

Reply:
Let's leave for now. Getting deleted in like < 5 days.